compiler/lib/source_map.mli

```diff
@@ -57,10 +59,16 @@ module Mappings : sig
       validation of its argument, unlike {!val:decode}. It is guaranteed that
       {!val:of_string} and {!val:to_string} are inverse functions. *)
 
-  val decode : t -> map list
+  val decode : t -> decoded
   (** Parse the mappings. *)
 
-  val encode : map list -> t
+  val encode : decoded -> t
+
+  val encode_with_offset : decoded -> int * t
```

Comment:
We should really document how this function differs from `encode` (IIUC it discards the first unmapped lines and returns their number).

Reply:
Done some more refactoring/renaming/documenting